operator/pkg/client/client_factory.go

```diff
@@ -103,11 +110,20 @@ func NewSingletonClients(config *rest.Config, logger logr.Logger) (*SingletonCli
 	discoveryRESTMapper := restmapper.NewDeferredDiscoveryRESTMapper(cachedDiscoveryClient)
 	discoveryShortcutExpander := restmapper.NewShortcutExpander(discoveryRESTMapper, cachedDiscoveryClient)
 
-	kubernetesClient, err := kubernetes.NewForConfigAndClient(config, httpClient)
+	// create runtime client only if not passed
+	runtimeClient := info.Client
+	if info.Client == nil {
+		runtimeClient, err = NewRuntimeClient(info.Config, discoveryShortcutExpander)
```

Comment:
This method here is not using the passed http Client and you will most likely have to proxy it to receive the correct logic. in `client.New` there are two clients created:
1. apiutil.NewDynamicRESTMapper(config)
2. metadata.NewForConfig(config)

For the first you will need to completely change to use a different mapper which you are doing already.
For the second you will have to override the client constructor which is internal since its not using `metadata.NewForConfigAndClient(config, client)`. We can discuss here what we should do instead.

Reply:
As discussed we will keep the implementation of having to http clients - one created inside `SingletonClients` and the other inside `client.New(),` since the typed client implementation inside the latter is not exposed via a public interface.